lib/actions/field-trip.js

```diff
@@ -0,0 +1,424 @@
+import { planParamsToQueryAsync } from '@opentripplanner/core-utils/lib/query'
+import { OTP_API_DATE_FORMAT, OTP_API_TIME_FORMAT } from '@opentripplanner/core-utils/lib/time'
+import moment from 'moment'
+import qs from 'qs'
+import { createAction } from 'redux-actions'
+
+import {routingQuery} from './api'
+import {setQueryParam} from './form'
+import {getGroupSize, getTripFromRequest, sessionIsInvalid} from '../util/call-taker'
+
+if (typeof (fetch) === 'undefined') require('isomorphic-fetch')
+
+/// PRIVATE ACTIONS
+
+const receivedFieldTrips = createAction('RECEIVED_FIELD_TRIPS')
+const requestingFieldTrips = createAction('REQUESTING_FIELD_TRIPS')
+const receivedFieldTripDetails = createAction('RECEIVED_FIELD_TRIP_DETAILS')
+const requestingFieldTripDetails = createAction('REQUESTING_FIELD_TRIP_DETAILS')
+
+// PUBLIC ACTIONS
+
+export const setFieldTripFilter = createAction('SET_FIELD_TRIP_FILTER')
+export const setActiveFieldTrip = createAction('SET_ACTIVE_FIELD_TRIP')
+export const setGroupSize = createAction('SET_GROUP_SIZE')
+export const toggleFieldTrips = createAction('TOGGLE_FIELD_TRIPS')
+
+/**
+ * Fetch all field trip requests (as summaries).
+ */
+export function fetchFieldTrips () {
+  return function (dispatch, getState) {
+    dispatch(requestingFieldTrips())
+    const {callTaker, otp} = getState()
+    if (sessionIsInvalid(callTaker.session)) return
+    const {datastoreUrl} = otp.config
+    const {sessionId} = callTaker.session
+    fetch(`${datastoreUrl}/fieldtrip/getRequestsSummary?${qs.stringify({sessionId})}`)
+      .then(res => res.json())
+      .then(fieldTrips => dispatch(receivedFieldTrips({fieldTrips})))
+      .catch(err => alert(`Error fetching field trips: ${JSON.stringify(err)}`))
```

Comment:
I believe this catch statement will also catch anything triggered by the dispatch call which maybe is ok, but might be hard to trace the problem to. A way to work around this is to rewrite this method to use async/await and then try/catch just the fetch call. This comment is non-blocking.